### ISSUE TITLE
Items label, state, and category values lost on server restart

### ISSUE THREAD

**Issue by jmichaluk** [eclipse/smarthome#2098]
Not all information is surviving a server outage. As shown below, the state, label, and category information is not being restored. The Paper UI also showed only a dimmer and no on/off switch after the restart.
I would have expected all information to be restored.

Before the server went down the Rest call for items returned something similar to the following: 

```
"link": "http://192.168.1.100:8888/rest/items/hue_0100_aaef5a30_1_brightness",
"state": "100",
"type": "Dimmer",
"name": "hue_0100_aaef5a30_1_brightness",
"label": "Brightness",
"category": "DimmableLight",
"tags": [
  "Lighting"
],
"groupNames": []
```

After the server came back up the Rest call returns:

```
"link": "http://192.168.1.100:8888/rest/items/hue_0100_efaabf2f_1_brightness",
"state": "NULL",
"type": "Dimmer",
"name": "hue_0100_efaabf2f_1_brightness",
"tags": [
  "Lighting"
],
"groupNames": []
```

The above was true for other items as well, and this has occurred multiple times.

**Comment by maggu2810** [comment]
How do you create such items?
Are the item created by the REST interface or do we talk about temporary created items because on item channel link is present?
The label, category, tags etc. should be stored for persistent items.
The state is not stored (if you use persistence and restoreOnStartup it could work but this is another topic) and IMHO it makes no sense to keep the old item state over a power loss (who knows if the state the ESH framework restores is still the state of the real pyhsical device's channel).

**Comment by kaikreuzer** [comment]
FTR: If the items are not manually defined, but inferred by the channel links, https://github.com/eclipse/smarthome/issues/1550 is likely to be the cause for the labels to disappear.

**Comment by jmichaluk** [comment]
This item was added when approving a discovery from the Hue Bridge.

**Comment by jmichaluk** [comment]
Steps to reproduce:

```
docker run \
       --name openhab \
       -v /etc/localtime:/etc/localtime:ro \
       -v /etc/timezone:/etc/timezone:ro \
       -v /home/umbrela/openhab/opt/openhab/conf:/openhab/conf \
       -v /home/umbrela/openhab/opt/openhab/userdata:/openhab/userdata \
       -p 8888:8080 \
       -p 8443:8443 \
       -p 5555:5555 \
       -d \
       --restart=always \
       openhab/openhab:amd64-online
```

- Add hue binding through paperUI
- Add hue bridge through paperUI 
- Add "Front Door" (hue Dimmable Light) through paperUI Inbox.

`docker restart openhab
`

**Comment by ruebox** [comment]
Dear all,

I am familiar in Java programming and also developing of openhab addon. However, I a new to openhab2 core. 
I.e. if you provide me some hints and support, I would be happy to contribute in fixing this issue.

**Comment by kaikreuzer** [comment]
Hi @ruebox, thanks for your interest to help on the core development!
For working on Eclipse SmartHome code, it is best to set up a dedicated IDE as described at https://www.eclipse.org/smarthome/documentation/development/ide.html. Alternatively, you can use the [openHAB IDE setup](http://docs.openhab.org/developers/development/ide.html) and also select the Eclipse SmartHome / Core option in step 4.

Please note that issue #2098 might not be a good choice to get started with, though, as it is highly complex. The root cause is very likely #1550 and this will probably best be tackled by #1896, where @sjka is currently trying to implement a solution for. So you might want to go for some other issue for a start :-)

**Comment by ruebox** [comment]
Hi @kaikreuzer thanks for the fast feeback ... I will checkout the currently open issues for smarthome. 
However, perhaps the bindings will be a good place for startup first ... 
Nevertheless, please keep me in mind if I can contribute to a specific topic.

**Comment by triller-telekom** [comment]
Since this is a very old issues and a lot of PRs have been merged since that I cannot tell which PR has fixed it but I just started ESH from my IDE, added a Hue bridge and a bulb and requested the state from an `Item` via the REST api:

```
http://localhost:8080/rest/items/hue_0210_001788141b41_1_color
```

led to:

```
{
link: "http://localhost:8080/rest/items/hue_0210_001788141b41_1_color",
state: "300,79,0",
type: "Color",
name: "hue_0210_001788141b41_1_color",
label: "Color",
category: "ColorLight",
tags: [
"Lighting"
],
groupNames: [ ]
}
```

after I restarted my ESH instance the same request led to:

```
{
link: "http://localhost:8080/rest/items/hue_0210_001788141b41_1_color",
state: "NULL",
type: "Color",
name: "hue_0210_001788141b41_1_color",
label: "Color",
category: "ColorLight",
tags: [
"Lighting"
],
groupNames: [ ]
}
```

So everything was persisted except for the state. Since the state cannot be guessed correctly it is safe to assume `NULL` as discussed above.

So I think this issue can be closed. @kaikreuzer @sjka Can you please close it?

**Comment by sjka** [comment]
On a retrospect, this likely got fixed by #3639
